src/classes/CRLP_Batch_Base.cls

```diff
@@ -344,4 +353,195 @@ public abstract class CRLP_Batch_Base {
+                            && rollup.Time_Bound_Operation_Type__c == CRLP_Operation.TimeBoundOperationType.Days_Back.name()
+                            && rollup.Integer__c != null) {
+                        if (maxDaysAgo == null || rollup.Integer__c > maxDaysAgo) {
+                            maxDaysAgo = (Integer) rollup.Integer__c;
+                        }
+                    }
+                }
+            }
+            return maxDaysAgo;
+        }
+        private set;
+    }
+
+    /**
+     * @description The max Integer for all active time-bound Years_Ago rollups,
+     * or null if there are no active time-bound Years_Ago rollups for this summary object
+     */
+    private Integer maxYearsAgo {
+        get {
+            if (maxYearsAgo == null) {
+                for (Rollup__mdt rollup : statefulCacheOfRollupsToBeProcessed) {
+                    if (rollup.Active__c == true
+                            && String.isNotBlank(rollup.Time_Bound_Operation_Type__c)
+                            && rollup.Time_Bound_Operation_Type__c == CRLP_Operation.TimeBoundOperationType.Years_Ago.name()
+                            && rollup.Integer__c != null) {
+                        if (maxYearsAgo == null || rollup.Integer__c > maxYearsAgo) {
+                            maxYearsAgo = (Integer) rollup.Integer__c;
+                        }
+                    }
+                }
+            }
+            return maxYearsAgo;
+        }
+        private set;
+    }
+
+    /**
+     * @description Determines whether the rollup should be run in incremental mode
+     * @param jobType the type of rollup job
+     * @return true if the rollup should be run in incremental mode, otherwise false
+     */
+    private Boolean isIncrementalMode() {
+        if (isScheduledJob == null || isScheduledJob == false) {
+            return false;
+        }
+
+        // Only run incremental mode when the custom settings are configured
+        // to enable incremental mode for this job type
+        Customizable_Rollup_Settings__c customizableRollupSettings = UTIL_CustomSettingsFacade.getCustomizableRollupSettings();
+        if (jobType == CRLP_RollupProcessingOptions.RollupType.AccountHardCredit
+                && customizableRollupSettings.AccountHardCreditNonSkew_Incremental__c == true) {
+            return true;
+        } else if (jobType == CRLP_RollupProcessingOptions.RollupType.ContactHardCredit
+                && customizableRollupSettings.ContactHardCreditNonSkew_Incremental__c == true) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * @description Produces the where clause that should be used for the inner join when in incremental mode
+     * @return The WHERE clause, as a string, that should be used to filter the detail records
+     */
+    protected String getIncrementalModeWhereClause() {
```

Comment:
This can be private since it's not used outside this class. I also recommend renaming this to `buildIncrementalModeWhereClause`.